### PR TITLE
Add an example gallery to the documentation

### PR DESCRIPTION
**Pull request by ntouran**
Add the beginnings of a example gallery to the ARMI documentation
using the sphinx-gallery plugin. This is intending to give new users a quick
view of the variety of things ARMI can do while also providing advanced users
with a quick reference to see how to do common operations.

### CODE REVIEW

File: armi/reactor/blocks.py [hunk 1]
```diff
@@ -1180,13 +1180,13 @@ def replaceBlockWithBlock(self, bReplacement):
         self.clearCache()
 
     @staticmethod
-    def plotFlux(core, fName, bList=None, peak=False, adjoint=False, bList2=None):
+    def plotFlux(core, fName=None, bList=None, peak=False, adjoint=False, bList2=None):
```
[the remark this answers is not in the document]
Comment (ntouran) [reply armi/reactor/blocks.py:1183]: yeah. I'll add a note to it saying as much. Also, this was used in a gallery example and then undone so we don't even need it for this. But I still like the minor cleanup.

File: armi/reactor/blocks.py [hunk 2]
```diff
@@ -1237,19 +1237,19 @@ def plotFlux(core, fName, bList=None, peak=False, adjoint=False, bList2=None):
...
             return
 
-        # write a little flux text file.
-        with open(title, "w") as f:
-            f.write("Energy_Group Average_Flux Peak_Flux\n")
-            for g, eMax in enumerate(lib.neutronEnergyUpperBounds):
-                f.write("{0} {1} {2}\n".format(eMax / 1e6, avg[g], peakFlux[g]))
+        if fName:
+            # write a little flux text file.
+            txtFileName = os.path.splitext(fName)[0] + ".txt"
```
[the remark this answers is not in the document]
Comment (ntouran) [reply armi/reactor/blocks.py:1248]: now the docstring does say so

File: armi/utils/plotting.py [hunk 3]
```diff
@@ -380,10 +380,24 @@ def plotFaceMap(
     else:
         plt.show()
 
-    plt.close()
+    # don't close figure here. Have caller call plotting.close or plt.close when
+    # they are done with it.
+
```
[the remark this answers is not in the document]
Comment (ntouran) [reply armi/utils/plotting.py:385]: I basically agree but did not go on that journey for this particular PR. 

A "clean up plotting stuff" sprint is probably called for at some point to help deal with this, and dealing with figure objects would be a great thing to put on that list. Ok I made #104 to track.

File: armi/utils/plotting.py [hunk 4]
```diff
@@ -380,10 +380,24 @@ def plotFaceMap(
...
+    Wrapper for matplotlib close. 
+
+    This is useful to avoid needing to import plotting and matplotlib.
+    The plot functions cannot always close their figure if it is going 
+    to be used somewhere else after becoming active (e.g. in reports
+    or gallery examples).
+    """
+    plt.close(fig)
+
+
```
[the remark this answers is not in the document]
Comment (ntouran) [reply armi/utils/plotting.py:400]: #104

File: armi/reactor/blocks.py [hunk 5]
```diff
@@ -1197,6 +1200,10 @@ def plotFlux(core, fName=None, bList=None, peak=False, adjoint=False, bList2=Non
         bList2 :
             a separate list of blocks that will also be plotted on a separate axis on the same plot.
             This is useful for comparing flux in some blocks with flux in some other blocks.
+
+        Notes
+        -----
+        This is not a great method. It should be cleand up and migrated into ``utils.plotting``.
```
Comment (youngmit) [inline armi/reactor/blocks.py:1206]: Perhaps a bit self-deprecating, haha. At some point, we should add a note at the top of `blocks.py` providing some of the context behind the method migrations. Beyond the scope of this changeset, though.

**Pull request merged**